src/server/p2p/network.rs

```diff
@@ -826,14 +827,17 @@ where S: ShareChain
             } else {
                 NUM_PEERS_TO_SYNC_PER_ALGO
             };
-            let peer_store_read_lock = self.network_peer_store.read().await;
-            let mut my_best_peers =
-                peer_store_read_lock.best_peers_to_share(num_peers, PowAlgorithm::RandomX, &request.known_peer_ids);
-            my_best_peers.extend(peer_store_read_lock.best_peers_to_share(
-                num_peers,
-                PowAlgorithm::Sha3x,
-                &request.known_peer_ids,
-            ));
+            let my_best_peers = {
+                let peer_store_read_lock = self.network_peer_store.read().await;
+                let mut my_best_peers =
+                    peer_store_read_lock.best_peers_to_share(num_peers, PowAlgorithm::RandomX, &request.known_peer_ids);
+                my_best_peers.extend(peer_store_read_lock.best_peers_to_share(
```

Comment:
the algo isn't actually used in this method so it can be changed to a single call